src/plugins/data/public/ui/query_editor/language_selector.tsx

```diff
@@ -44,7 +44,7 @@ export const QueryLanguageSelector = (props: QueryLanguageSelectorProps) => {
     : undefined;
 
   useEffect(() => {
-    const subscription = queryString.getUpdates$().subscribe((query: Query) => {
+    const subscription = queryString.getUpdates$(0).subscribe((query: Query) => {
```

Comment:
is it potentially just a race condition or something?

could we verify if the props.query.language is the current query always. and this component gets re-rendered. i need to verify since last checking these things out. or if you have the insight already.

because what im thinking is that potentially the dataset is checking which languages are supported and adding to a list while this is listening to changes and before the language is added this setter is called but set to the first language it found.

there's a couple of places i think were we need loading spinners. shooting in the dark perhaps we just need to check if everything is done loading before setting the current language?

Reply:
or opt for just disabling unsupported languages and setting the current value vs removing them from view